workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Readers/ReaderBase.cs

```diff
@@ -0,0 +1,154 @@
+
+        public event GameObjectDelegates.ComponentUpdated<TComponentUpdate> ComponentUpdated
+        {
+            add => componentUpdateDelegates.Add(value);
+            remove => componentUpdateDelegates.Remove(value);
+        }
+
+        /// <summary>
+        /// Helper method to dispatch property updates to callbacks while forwarding exceptions to the log dispatcher.
+        /// </summary>
+        /// <param name="payload">The value for the property update.</param>
+        /// <param name="callbacks">The property update handlers.</param>
+        /// <typeparam name="T">The property update type.</typeparam>
+        protected void DispatchWithErrorHandling<T>(Option<T> payload, IEnumerable<Action<T>> callbacks)
+        {
+            if (!payload.HasValue)
+            {
+                return;
+            }
+
+            foreach (var callback in callbacks)
+            {
+                try
+                {
+                    callback(payload.Value);
+                }
+                catch (Exception e)
+                {
+                    // Log the exception but do not rethrow it, as other delegates should still get called
+                    logDispatcher.HandleLog(LogType.Exception, new LogEvent().WithException(e));
+                }
+            }
+        }
+
+        public void OnAuthorityChange(Authority authority)
+        {
+            foreach (var callback in authorityChangedDelegates)
+            {
+                try
+                {
+                    callback(authority);
+                }
+                catch (Exception e)
+                {
+                    // Log the exception but do not rethrow it, as other delegates should still get called
+                    logDispatcher.HandleLog(LogType.Exception, new LogEvent().WithException(e));
+                }
+            }
+        }
+
+        public void OnComponentUpdate()
+        {
+            foreach (var update in EntityManager.GetComponentObject<ComponentsUpdated<TComponentUpdate>>(Entity).Buffer)
+            {
+                foreach (var callback in componentUpdateDelegates)
+                {
+                    try
+                    {
+                        callback(update);
+                    }
+                    catch (Exception e)
+                    {
+                        // Log the exception but do not rethrow it, as other delegates should still get called
+                        logDispatcher.HandleLog(LogType.Exception, new LogEvent().WithException(e));
+                    }
+                }
+
+                TriggerFieldCallbacks(update);
+            }
+        }
+
+        /// <summary>
+        ///     Reader implementations will override this if their components have fields.
+        /// </summary>
+        /// <param name="update"></param>
+        protected virtual void TriggerFieldCallbacks(TComponentUpdate update)
+        {
+        }
+
+        public void OnEvent(int eventIndex)
```

Comment:
Similarly, events and commands should have separate methods with appropriate parameters

Reply:
they will be taken care of in other PRs

Reply:
Now that you did add codegening the separate methods though, you might as well remove this